app/scripts/controllers/transactions/index.js

```diff
@@ -694,6 +695,45 @@ export default class TransactionController extends EventEmitter {
     return this._getTransaction(txId);
   }
 
+  transactionHasGasDefaults(txMeta) {
+    return txMeta.txParams.gasPrice || txMeta.txParams.maxPriorityFeePerGas;
+  }
+
+  async addTransactionGasDefaults(txMeta) {
+    if (this.transactionHasGasDefaults(txMeta)) {
+      return txMeta;
+    }
+
+    const contractCode = await determineTransactionContractCode(
+      txMeta.txParams,
+      this.query,
+    );
+
+    let updateTxMeta = txMeta;
+    try {
+      updateTxMeta = await this.addTxGasDefaults(txMeta, contractCode);
+    } catch (error) {
+      log.warn(error);
+      updateTxMeta = this.txStateManager.getTransaction(txMeta.id);
+      updateTxMeta.loadingDefaults = false;
+      this.txStateManager.updateTransaction(
+        txMeta,
+        'Failed to calculate gas defaults.',
+      );
+      throw error;
+    }
+
+    updateTxMeta.loadingDefaults = false;
+
+    // save txMeta
+    this.txStateManager.updateTransaction(
+      updateTxMeta,
+      'Added new unapproved transaction.',
+    );
```

Comment:
Why was this moved out of the addUnapprovedTransaction method? Instead we call into another method that then saves the transaction? I feel like we're adding some misdirection here.

Reply:
It appears to me that this update is called only to save default gas estimates. In fact the note `Added new unapproved transaction.'` is confusing.

Reply:
Okay, fair assessment, it looks like the addTransaction method that gets called doesn't have a log statement so the previous contributors relied upon this update to add the log for the add transaction part of the history.

Reply:
Let me add comment here and also create an issue to fix this later.

Reply:
> +    // save txMeta
+    this.txStateManager.updateTransaction(
+      updateTxMeta,
+      'Added new unapproved transaction.',
+    );